Tests/Unit Tests/Conditions/CreditCardConditionTests.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  CreditCardConditionTests.swift
+//  FormValidatorSwift
+//
+//  Created by Onur Ersel on 2016-11-02.
+//  Copyright © 2016 ustwo. All rights reserved.
+//
+
+import XCTest
+
+@testable import FormValidatorSwift
+
+
+final class CreditCardConditionTests: XCTestCase {
+    
+    
+    // MARK: - Properties
+    
+    // Conditions
+    let conditionAll                = CreditCardCondition()
+    let conditionAmericanExpress    = CreditCardCondition(cardType: .americanExpress)
+    let conditionDinersClub         = CreditCardCondition(cardType: .dinersClub)
+    let conditionDiscover           = CreditCardCondition(cardType: .discover)
+    let conditionJcb                = CreditCardCondition(cardType: .jcb)
+    let conditionMaestro            = CreditCardCondition(cardType: .maestro)
+    let conditionMasterCard         = CreditCardCondition(cardType: .mastercard)
+    let conditionVisa               = CreditCardCondition(cardType: .visa)
+    
+    // Valid Card Numbers
+    let cardNumberAmericanExpress   = "376031710126369"
+    let cardNumberDinersClub        = "30085182354725"
+    let cardNumberDiscover          = "6011359046468736"
+    let cardNumberJcb               = "3535983484092395"
+    let cardNumberMaestro           = "6762856858323942"
+    let cardNumberMastercard        = "5480785928215247"
+    let cardNumberVisa              = "4024007127428075"
+    
+    let cardNumberWhitespace        = "3760 3171  0126       369"
+    
+    // Invalid Card Numbers
+    let cardNumberInvalid_1         = "0000000000000000"
+    let cardNumberInvalid_2         = "67628 568"
+    
+    
+    
+    // MARK: - Test Success
+    
+    func testCreditCardCondition_Success() {
+        // Given
+        let expectedResult = true
+        
+        // Test
+        AssertCondition(conditionAll, testInput: cardNumberAmericanExpress, expectedResult: expectedResult)
```

Comment:
Again, you appear to have a lot of repeated code here. Perhaps we can put some of these into an array and loop through the array with an assert on each member of the array? It'll allow expanding the number of credit cards we can validate in the future.